electron.js

```diff
@@ -1,38 +1,251 @@
 const electron = require('electron');
+const { app, Menu, MenuItem, BrowserWindow, dialog, ipcMain} = require('electron');
 
-// Module to control application life.
-const app = electron.app;
-// Module to create native browser window.
-const BrowserWindow = electron.BrowserWindow;
+// Module to create handle file management
+const fs = require('fs');
 
-// Keep a global reference of the window object, if you don't, the window will
-// be closed automatically when the JavaScript object is garbage collected.
-let mainWindow;
+// Module to handle window management
+const windowManager = require('electron-window-manager');
+var numWindows = 0;
+var windowNameStart = 'Window #';
+
+const template = [
```

Comment:
Where did all this code come from / what does it do? Do we need this?

Reply:
numWindows is incremented each time a new window is created, which is then used with windowNameStart to create a unique window name for each BrowserWindow instance, which is what allows us to create and manage separate browser windows and have actions apply to only specific windows. The template is used for top menu options. I think it should be left in, though I'm planning a refactor in the next week or so.